core-bundle/src/Controller/ContentElement/AbstractContentElementController.php

```diff
@@ -37,6 +42,12 @@ public function __invoke(Request $request, ContentModel $model, string $section,
             $response = $template->getResponse();
         }
 
+        $time = time();
+        if (!$response->headers->has('Cache-Control') && '' !== $model->stop && $model->stop > $time) {
+            $response->setPublic();
+            $response->setMaxAge((int) $model->stop - $time);
+        }
+
```

Comment:
IMHO we should either put this in a protected method (`getExpiredResponse`?) as well or also inline `getUpcomingResponse`. I don't think it's a typical use case to disable this logic because the design of Contao content elements implies a start/stop setting; so I would vote for inlining. This would also prevent a glitch if the two `time()` calls produce a different output right at the edge of the stop time. 

If needed we could add a property `protected bool $handleStartStop = true` and check for that then. Wdyt?

Reply:
Btw. we discussed this, but I forgot the outcome: Why don't we use the `ResponseCacheStrategy` to merge the max age? IMHO this should work as intended, because it's no different than adding another content element with a limiting max age to the same page. This way an element could still set cache headers (like "this response generated from an API is cacheable for 30min") and the stop setting would still work 10min before it is hit. Otherwise you always have to consider the stop time yourself when constraining the max age.